src/devtools/client/debugger/src/reducers/breakpoints.ts

```diff
@@ -371,3 +381,120 @@ export function getLogpointsForSource(state: UIState, sourceId: string) {
+  b: LocationAnalysisPoints | undefined
+) => {
+  const d1 = a?.data ?? [];
+  const d2 = b?.data ?? [];
+
+  let dataEqual =
+    d1.length === d2.length &&
+    d1.every((item, i) => {
+      const otherItem = d2[i];
+      const pointEqual = item.point === otherItem.point;
+      const timeEqual = item.time === otherItem.time;
+      return pointEqual && timeEqual;
+    });
+  const errorEqual = a?.error === b?.error;
+  const result = dataEqual && errorEqual;
+
+  return result;
+};
+
+export const getAnalysisPointsForLocation = createSelector(
+  [
+    (state: UIState, location: Location | null) => {
+      if (!location) {
+        return undefined;
+      }
+      const locationKey = getLocationKey(location);
+      return state.breakpoints.analysisMappings.entities[locationKey];
+    },
+    (state: UIState) => state.breakpoints.analyses,
+    (state: UIState) => state.timeline.focusRegion,
+
+    (state: UIState, location: Location | null, condition: string | null = "") => condition,
+  ],
+  (mappingEntry, analyses, focusRegion, condition): LocationAnalysisPoints | undefined => {
+    // First, verify that we have a real location and a breakpoint mapping for that location
+    if (!mappingEntry) {
+      return undefined;
+    }
+
+    const matchingEntries: AnalysisSummary[] = [];
+
+    // Next, filter down all analysis runs for this location, based on matching
+    // against the `condition` the user supplied, as well as whether we
+    // actually legitimately found points
+    mappingEntry.allAnalyses.forEach(analysisId => {
+      const analysisEntry = analyses.entities[analysisId];
+      // TODO Double-check undefined vs empty string conditions here
+      if (!analysisEntry || (analysisEntry.condition ?? "") !== condition) {
+        return;
+      }
+
+      // TODO Are we sure about the error statuses here?
+      const hasPoints = [
+        // Successful queries
+        AnalysisStatus.PointsRetrieved,
+        AnalysisStatus.Completed,
+        // Presumably got points first
+        AnalysisStatus.LoadingResults,
+        // Should have at least come back with _some_ points
+        AnalysisStatus.ErroredGettingPoints,
+        AnalysisStatus.ErroredRunning,
+      ].includes(analysisEntry.status);
+
+      if (hasPoints) {
+        matchingEntries.push(analysisEntry);
+      }
+    });
+
+    if (matchingEntries.length === 0) {
+      // We have no hits available
+      return undefined;
+    }
+
+    const allPoints = matchingEntries.map(entry => entry.points!).flat();
+    const uniquePoints = uniqBy(allPoints, item => item.point);
+    uniquePoints.sort((a, b) => compareNumericStrings(a.point, b.point));
+
+    const finalPoints = focusRegion ? filterToFocusRegion(uniquePoints, focusRegion) : uniquePoints;
+
+    // Meanwhile, we're going to use the _latest_ analysis run's status
+    // for display purposes.
+    const latestAnalysisEntry = analyses.entities[mappingEntry.currentAnalysis!];
```

Comment:
I think it would also be *a-okay* to have `getAnalysisStatusForLocation` and `getAnalysisPointsForLocation`, but again, could be a follow-on.

Reply:
https://app.replay.io/recording/false-condition-does-not-hide-logpoints--f74ebd50-d961-4c96-b996-5711a4112ed1?point=80805119884749729135803403855724710&time=31659.124300593958&hasFrames=true

Reply:
I wonder if we should keep the concept of an `analysisRequested` action, which clears the `currentAnalysis` for that location?